File: routes/articles.py
from fastapi import APIRouter, Request
from fastapi.responses import HTMLResponse
from fastapi.templating import Jinja2Templates
from models.article import Article
from database import SessionLocal
import markdown

from services.article_admin import get_article, get_all_articles

router = APIRouter(prefix="/articles", tags=["Articles Public"])
templates = Jinja2Templates(directory="templates")


@router.get("/{article_id}", response_class=HTMLResponse)
def view_article(request: Request, article_id: int):
    db = SessionLocal()
    article = db.query(Article).get(article_id)
    db.close()

    if not article:
        return HTMLResponse("<h1>Article introuvable</h1>", status_code=404)

    # Convertir le contenu Markdown en HTML
    html_content = markdown.markdown(article.content, extensions=["fenced_code", "tables", "toc"])

    return templates.TemplateResponse(
        "article_detail.html",
        {"request": request, "article": article, "html_content": html_content}
    )

File: routes/articles.js
// routes/articles.js
const express = require('express');
const router = express.Router();
const { getArticle } = require('../services/article_admin');
const Article = require('../models/Article');
const { marked } = require('marked');

// Configuration de marked pour les extensions
marked.setOptions({
  breaks: true,
  gfm: true
});

router.get('/:articleId', (req, res) => {
  const articleId = parseInt(req.params.articleId);
  const article = getArticle(articleId);
  
  if (!article) {
    return res.status(404).send('<h1>Article introuvable</h1>');
  }
  
  // Convertir le contenu Markdown en HTML
  const htmlContent = marked(article.content);
  
  res.render('article_detail', { article, html_content: htmlContent });
});

module.exports = router;

